tests/x11/firefox/firefox_html5.pm

```diff
@@ -25,7 +25,7 @@ sub run {
     x11_start_program('xterm');
     script_run('cd ~/data/testwebsites');
     enter_cmd('python3 -m http.server 48080 &');
-    assert_script_run 'curl --connect-timeout 5 --max-time 10 --retry 5 --retry-delay 0 --retry-max-time 40 http://localhost:48080/';
+    assert_script_run 'curl --connect-timeout 5 --max-time 10 --retry-connrefused 5 --retry-delay 1 --retry-max-time 40 http://localhost:48080/';
```

Comment:
Can you add a comment, that if it fails on in longer than 20secs, we need to reevaluate how the python webserver is being started?

Reply:
Uh :O I did not know that `curl` implements `retry` internally.

Reply:
I have a minor comment, can you please add a port check with `script_retry`?
ss -tnlp | grep 48080

IMO, if the port is up, then curl should work :)

Reply:
That's a valid point, @rfan1, but apart from @pdostal 's comment, I think that if the server is not up in 10-15 seconds for whatever reason it's a good idea to fail and the reason will be visible in the results. At this point I think we should leave this as it is. If this ends up happening often, we can re-evaluate.

Reply:
@foursixnine I added a comment to explain that the server is given 10-15 seconds to come up. Exact time is not available, since this is dependent on the reason of failure and the time it takes for curl to fail, which is variant. Curl here tries to connect the first time for 5 seconds, then retries 5 times, with 1 second between tries, but how long it will take each try to fail is variant and reason-dependent.

It could potentially not fail even in 20 seconds (if curl waited for all the 5 second period each time) but here it won't, because curl fails quickly with "connection refused".

Now that I think about it, I will remove all exact time numbers from the comment.